ocw/lib/EC2.py

```diff
@@ -77,6 +78,38 @@ def all_clusters(self):
             [clusters.append(cluster) for cluster in response['clusters']]
         return clusters
 
+    @staticmethod
+    def needs_to_delete_snapshot(snapshot, cleanup_ec2_max_snapshot_age_days) -> bool:
+        delete_older_than = date.today() - timedelta(days=cleanup_ec2_max_snapshot_age_days)
+        if datetime.date(snapshot['StartTime']) < delete_older_than:
+            regexes = [
+                re.compile(r'''^OpenQA upload image$'''),
+                re.compile(r'''^Created by CreateImage\([\w-]+\) for ami-\w+ from vol-\w+$''')
+            ]
+            for regex in regexes:
+                m = re.match(regex, snapshot['Description'].strip())
+                if m:
+                    return True
+        return False
+
+    def cleanup_snapshots(self):
+        cleanup_ec2_max_snapshot_age_days = self.cfgGet('cleanup', 'ec2-max-snapshot-age-days')
+        if cleanup_ec2_max_snapshot_age_days > 0:
```

Comment:
This function has only one path to succeed. I don't see a good reason why not return on the faulty path and avoid with it deep indention. 
```suggestion
        if cleanup_ec2_max_snapshot_age_days < 1:
            return
```

Reply:
As the default is `-1` I would actually treat 0 as delete snapshots immediatly.
So this would be my take then:
```suggestion
        if cleanup_ec2_max_snapshot_age_days < 0:
            return
```